skimage/draw/_random_shapes.py

```diff
@@ -200,13 +205,34 @@ def _generate_random_colors(num_colors, num_channels, intensity_range, random):
         each channel are drawn from the corresponding `intensity_range`.
 
     """
+
     if num_channels == 1:
-        intensity_range = (intensity_range, )
+        intensity_range_tpl = (intensity_range, )
     elif len(intensity_range) == 1:
-        intensity_range = intensity_range * num_channels
+        intensity_range_tpl = intensity_range * num_channels
+    else:
+        intensity_range_tpl = intensity_range
+
+    if np.diff(np.array(intensity_range_tpl)).sum() == 0:
+        # if intensity range only spans a single color
+        colors = np.array([np.array(intensity_range_tpl)[:, 0]])
```

Comment:
To many `np.array` constructors here, above, and below. Please, cast `intensity_range_tpl` to ndarray as early as possible, and use the vectorized ops.

Reply:
Sorry to interject, but `asarray` might be appropriate to avoid a copy early on.

Also, are you looking for `np.newaxis`. you would slice with `[:, 0, np.newaxis]`

Reply:
@hmaarrfk thanks for the hint!